tests/test_connection_config.py

```diff
@@ -0,0 +1,69 @@
+import os
+import pytest
+
+from snowcli.connection_config import ConnectionConfigs
+from snowcli.exception import EnvironmentVariableNotFoundError
+from unittest import mock
+
+CONFIG_PATH = "tests/config/env_variables_config"
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "SNOWCLI_ACCOUNT_NAME": "account_name",
+        "SNOWCLI_USER": "username",
+        "SNOWCLI_PASS": "password",
+        "SNOWCLI_HOST": "localhost",
+        "SNOWCLI_PORT": "8080",
+        "SNOWCLI_PROTOCOL": "http",
+    },
+)
+def test_replace_with_env_variables():
+    result = ConnectionConfigs(CONFIG_PATH)
+
+    assert result.get_connection("test") == {
+        "account": "account_name",
+        "user": "username",
+        "password": "password",
+        "host": "localhost",
+        "port": "8080",
+        "protocol": "http",
+    }
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "SNOWCLI_USER": "username",
+        "SNOWCLI_PASS": "password",
+        "SNOWCLI_HOST": "localhost",
+        "SNOWCLI_PORT": "8080",
+        "SNOWCLI_PROTOCOL": "http",
+    },
+)
+def test_missing_account_in_environment_variables():
+    result = ConnectionConfigs(CONFIG_PATH)
+
+    with pytest.raises(EnvironmentVariableNotFoundError):
```

Comment:
IMO we should use exceptions not related to ClickException only if sth really unexpected happens but I don't even have an idea what it can be. IMO such exceptions will be thrown by external libraries.

In all other cases I would try to choose one of exceptions from `ClickException` or create own subclass of it (but only if it will give any benefit). Using `ClickException` allows Click to handle an exception and convert it to a appropriate result code and nicely format the error message. 

See `test_sql_fails_for_both_stdin_and_other_query_source` for an example of test presenting the difference (exception isn't propagated outside the framework). You can also try to execute both cases on your own and see actual difference in output visible in your terminal.

WDYT?

Reply:
I agree. `EnvironmentVariableNotFoundError` extends `ClickException` now